### ISSUE TITLE
Benchmarks for MIMIC-CXR dataset

### ISSUE THREAD

**Issue by santosh9sanjeev** [mlmed/torchxrayvision#144]
Hi,
Great work on exploring different datasets and releasing the pretrained weight. But I cannot find the results on the MIMIC-CXR dataset. Can you pls let me know where to find them.

Thanks!

**Comment by ieee8023** [comment]
Oh, sorry I didn't realize they weren't there. I'll launch some jobs to compute them in the coming days and get back to you.

**Comment by santosh9sanjeev** [comment]
Sure also, can you please follow the official split, as it will be easier to croosverify it. I observe that in the model_calibrate.py file it does a random split.
That would be really helpful for me.
Thank you!

**Comment by santosh9sanjeev** [comment]
Hi 
I also had few doubts.
1) The mimic pretrained model the "densenet121-res224-mimic_ch", how come has only 11 pathologies and not 13 as per the actual dataset.
2) How do you ignore the -1 classes? Do you consider them as 0?
3) Lets say I have to run a pretrained model on MIMIC-CXR test set, I follow the code in the github main page. It gives really bad results. Am I missing something ? Also it gives only probabilities, how should I select the best optimal threshold or is it preselected based on the op_thresh?

Please let me know.

Thanks!

**Comment by ieee8023** [comment]
> 1. The mimic pretrained model the "densenet121-res224-mimic_ch", how come has only 11 pathologies and not 13 as per the actual dataset.

The two missing are "Fracture" "Support Devices" that were removed because they were not very common across datasets used in the work that trained all the models: https://arxiv.org/abs/2002.02497

> 2. How do you ignore the -1 classes? Do you consider them as 0?

The datasets encode unknowns as nans https://github.com/mlmed/torchxrayvision/blob/26db733796be0caa862933bdd840717a0ded2a68/torchxrayvision/datasets.py#L1054
then during training the these are not included in the loss: https://github.com/mlmed/torchxrayvision/blob/26db733796be0caa862933bdd840717a0ded2a68/scripts/train_utils.py#L177

> 3. Lets say I have to run a pretrained model on MIMIC-CXR test set, I follow the code in the github main page. It gives really bad results. Am I missing something ? Also it gives only probabilities, how should I select the best optimal threshold or is it preselected based on the op_thresh?

Perhaps the images are not normalized correctly. Have you see the process image example that will read in image files? https://github.com/mlmed/torchxrayvision/blob/master/scripts/process_image.py

> Also it gives only probabilities, how should I select the best optimal threshold or is it preselected based on the op_thresh?

The pretrained models already have the op_thresh applied. They are calibrated on a hold out test set from the same dataset.

Do you want to post the results you obtained? I'm still working on downloading the MIMIC data so I can run the benchmark. You can find AUC values computed in the original paper in Figure 1 here: https://arxiv.org/abs/2002.02497

**Comment by ieee8023** [comment]
I added benchmarks for the MIMIC dataset. The chexpert classifier will take another day to process: https://github.com/mlmed/torchxrayvision/blob/master/BENCHMARKS.md#mimic-cxr-chexpert-labels

**Comment by ieee8023** [comment]
> use the model_calibrate file for testing, which is similar to a test script.

This is the same script I am using the generate the benchmarks.

> I used the official test split given by MIMIC-CXR. It has around 5159 test samples.

The test set used by that script is here. These are the indices into the dataset of just ["PA","AP"] views which has the length 243324. The test set has 48456 samples  [mimic_xrv_testset.csv](https://github.com/mlmed/torchxrayvision/files/12820903/mimic_xrv_testset.csv)

The model was trained using this split to hold out test samples, so the official MIMIC test samples are mixed in.

> I used the opt_thresh on the test set only. (So the opt_thresh is calculated on the test set and used on it, though the general procedure is to calculate the opt_thresh on the validation set and then use it on the test set. I tried that too but obviously the results are lower than the former.)

The opt_thresh won't change the AUC because you don't need to select a threshold. It will impact the F1 and acc though. 

The results you post don't look that bad in terms of AUC. This seems on par with the reality of these models. The acc can be misleading because the classes are not balanced.

**Comment by santosh9sanjeev** [comment]
Got it! Thank you so much!
Just to confirm, so you have used the train set in MIMIC which has 243334 of AP and PA views(matches with the actual csv file also only ['PA,AP']. U have split this into train and test right? and then tested on the test set.
Pls correct me if I am wrong.

Thank you 😃

**Comment by ieee8023** [comment]
Yes exactly. And when training a model on multiple datasets (as is done with the 'all' model) it will work the same way for each dataset and it will take out a train and test split and then merge all the train sets together as seen here:

https://github.com/mlmed/torchxrayvision/blob/e62ce3ca0a503db5964c1bd07cbea6c2650e4957/scripts/train_model.py#L145-L158

Inside the training step it will cut out a validation set from the main training split.

One of the goals of this library is to remove the headache of tracking down which split to evaluate on and making it easy to train on a truly external dataset by having the interface to each dataset be the same. I prefer to report results on a dataset (or datasets) that was not used when creating the model at all.

**Comment by santosh9sanjeev** [comment]
Got it!
Thank you so much for clearing my doubts! 😄 

I am closing the issue as its sorted out.
